documentation/assets/Example: training an external model.md

```diff
@@ -0,0 +1,753 @@
+    (dec2relu2): ReLU(inplace=True)
+  )
+  (upconv1): ConvTranspose2d(64, 32, kernel_size=(2, 2), stride=(2, 2))
+  (decoder1): Sequential(
+    (dec1conv1): Conv2d(64, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec1norm1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec1relu1): ReLU(inplace=True)
+    (dec1conv2): Conv2d(32, 32, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
+    (dec1norm2): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
+    (dec1relu2): ReLU(inplace=True)
+  )
+  (conv): Conv2d(32, 1, kernel_size=(1, 1), stride=(1, 1))
+)
+```
+
+## 3. Loss function
+
+While a popular choice for binary segmentation loss function is the binary cross-entropy (BCE) loss, often it is
+the intersection-over-union (IoU), or the [Jaccard Index](https://en.wikipedia.org/wiki/Jaccard_index), that serves
+as a measure of success. In this example, we will use a combination of the BCE and IoU as the loss function. While
+SuperGradients provides, among many other losses, an implementation of the combined BCE and Dice loss, which could be 
+used for our purposes as well, we will show how we can define our own user-defined loss function and train our model 
+with it using SuperGradients.
+
+Similar to using an external model, the custom loss function's class must inherit from `torch.nn.Module`. The 
+`forward()` function's first parameter needs to be the predictions tensor and the second parameter needs to be the
+targets tensor.
+
+```
+import torch
+import torch.nn as nn
+from torchmetrics import JaccardIndex
+
+
+class CustomSegLoss(torch.nn.Module):
+    def __init__(self, bce_weight=1, iou_weight=1):
+        super(CustomSegLoss, self).__init__()
+
+        self.bce_weight = bce_weight
+        self.iou_weight = iou_weight
+
+        self.bce_loss = nn.BCELoss()
+        self.iou_func = JaccardIndex(num_classes=2)
+
+    def forward(self, preds, target):
+
+        bce_loss = self.bce_loss(preds, target)
+        iou_loss = 1.0 - self.iou_func(torch.gt(preds, 0.5).long(), torch.gt(target, 0.5).long())
+
+        return self.bce_weight*bce_loss + self.iou_weight*iou_loss
+```
+
+Notice that here the BCE loss term is obtained simply by using PyTorch's `BCELoss()`. To compute the IoU score, or
+the Jaccard Index, we use torchmetric's `JaccardIndex`. To compute the IoU, the pixel values in both images should
+be binary, i.e., 0's and 1's. Since in the model's forward function a sigmoid function is already applied to the
+output, we only need to binarize both the predictions and the target tensors with a threshold of 0.5 (remember, we are 
+using **soft** masks). When measuring segmentation performance, higher IoU is better. Since IoU score is a number 
+in [0, 1], we simply compute `IoU loss = 1 - IoU` to make it a valid loss function for gradient-descent optimization.
```

Comment:
If you decide to drop the ious component make sure to fix this as well.

Reply:
no need, kept the IoU